core/src/mender-client.c

```diff
@@ -1165,17 +1009,30 @@ mender_client_update_work_function(void) {
         goto END;
     }
 
+    bool only_for_testing = true;
+
+    mender_log_info("Getting device type");
+
     /* Retrieve device type from artifact */
     const char *device_type_artifact = NULL;
     if (MENDER_OK != (ret = mender_artifact_get_device_type(mender_artifact_ctx, &device_type_artifact))) {
         mender_log_error("Unable to get device type from artifact");
-        mender_client_publish_deployment_status(deployment->id, MENDER_DEPLOYMENT_STATUS_FAILURE);
-        if (mender_client_deployment_needs_set_pending_image) {
-            mender_flash_abort_deployment(mender_client_flash_handle);
+        if (true == only_for_testing) {
```

Comment:
No need to compare boolean value with `true`

Reply:
Actually, I have to remove this. Thanks

Reply:
(I mean, you can see that it is just set above to `true` and never modified...`